### ISSUE TITLE
Error in Encoder

### ISSUE THREAD

**Issue by proska** [usc-isi-i2/dsbox-ta2#109]
Running Command:
```
python ta2-search ~/dsbox/runs2/config-seed-test/LL0_40687_solar_flare_2/search_config.json
```

Error Message:
```
Traceback (most recent call last):
  File "/nfs1/dsbox-repo/qasemi/dsbox-ta2/python/dsbox/template/search.py", line 523, in evaluate_pipeline
    evaluation_result = self._evaluate(configuration, cache, dump2disk)
  File "/nfs1/dsbox-repo/qasemi/dsbox-ta2/python/dsbox/template/search.py", line 546, in _evaluate
    fitted_pipeline.fit(cache=cache, inputs=[self.train_dataset1])
  File "/nfs1/dsbox-repo/qasemi/dsbox-ta2/python/dsbox/pipeline/fitted_pipeline.py", line 94, in fit
    self.runtime.fit(**arguments)
  File "/nfs1/dsbox-repo/qasemi/dsbox-ta2/python/dsbox/template/runtime.py", line 195, in fit
    primitive_arguments
  File "/nfs1/dsbox-repo/qasemi/dsbox-ta2/python/dsbox/template/runtime.py", line 284, in _primitive_step_fit
    produce_result = model.produce(**produce_params)
  File "/nfs1/dsbox-repo/qasemi/dsbox-cleaning/dsbox/datapreprocessing/cleaner/encoder.py", line 194, in produce
    self._input_data_copy = utils.remove_columns(self._input_data_copy, drop_indices, source='ISI DSBox Data Encoder')
  File "/nfs1/dsbox-repo/qasemi/common-primitives/common_primitives/utils.py", line 349, in remove_columns
    raise ValueError("Removing columns would have removed the last column.")
ValueError: Removing columns would have removed the last column.
```

Full Output:
```
[INFO] - dsbox.controller.controller - [INFO] Template 1:Default_classification_template Selected. UCT:[None, None, None, None, None, None, None, None]
[INFO] Worker started, id: <_MainProcess(MainProcess, started)> , True
/nfs1/dsbox-repo/qasemi/miniconda/envs/d3m-devel/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.
[INFO] Will use cross validation( n = 10 ) to choose best primitives.
[INFO] Push@cache: ('d3m.primitives.dsbox.Denormalize', 8230156918274372836)
/nfs1/dsbox-repo/qasemi/miniconda/envs/d3m-devel/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.
[INFO] Push@cache: ('d3m.primitives.datasets.DatasetToDataFrame', 8230156918274372836)
[INFO] Push@cache: ('d3m.primitives.data.ExtractColumnsBySemanticTypes', -4501480681699038751)
[INFO] Push@cache: ('d3m.primitives.data.ExtractColumnsBySemanticTypes', -650282118701789208)
[INFO] Push@cache: ('d3m.primitives.dsbox.Profiler', -1673142002480840049)
[INFO] Push@cache: ('d3m.primitives.dsbox.CleaningFeaturizer', -1673142002480840049)
[INFO] Push@cache: ('d3m.primitives.dsbox.CorexText', -1673142002480840049)
[INFO] Push@cache: ('d3m.primitives.dsbox.Encoder', -1673142002480840049)
Traceback (most recent call last):
  File "/nfs1/dsbox-repo/qasemi/dsbox-ta2/python/dsbox/template/search.py", line 523, in evaluate_pipeline
    evaluation_result = self._evaluate(configuration, cache, dump2disk)
  File "/nfs1/dsbox-repo/qasemi/dsbox-ta2/python/dsbox/template/search.py", line 546, in _evaluate
    fitted_pipeline.fit(cache=cache, inputs=[self.train_dataset1])
  File "/nfs1/dsbox-repo/qasemi/dsbox-ta2/python/dsbox/pipeline/fitted_pipeline.py", line 94, in fit
    self.runtime.fit(**arguments)
  File "/nfs1/dsbox-repo/qasemi/dsbox-ta2/python/dsbox/template/runtime.py", line 195, in fit
    primitive_arguments
  File "/nfs1/dsbox-repo/qasemi/dsbox-ta2/python/dsbox/template/runtime.py", line 284, in _primitive_step_fit
    produce_result = model.produce(**produce_params)
  File "/nfs1/dsbox-repo/qasemi/dsbox-cleaning/dsbox/datapreprocessing/cleaner/encoder.py", line 194, in produce
    self._input_data_copy = utils.remove_columns(self._input_data_copy, drop_indices, source='ISI DSBox Data Encoder')
  File "/nfs1/dsbox-repo/qasemi/common-primitives/common_primitives/utils.py", line 349, in remove_columns
    raise ValueError("Removing columns would have removed the last column.")
ValueError: Removing columns would have removed the last column.
[INFO] push@Candidate: (-3485808742739620201,None)
Traceback (most recent call last):
  File "/nfs1/dsbox-repo/qasemi/dsbox-ta2/python/dsbox/template/search.py", line 378, in setup_initial_candidate
    candidate.data.update(result)
TypeError: 'NoneType' object is not iterable
[ERROR] Initial Pipeline failed, Trying a random pipeline ...
{'clean_step': {'hyperparameters': {},
                'primitive': 'd3m.primitives.dsbox.CleaningFeaturizer'},
 'corex_step': {'hyperparameters': {},
                'primitive': 'd3m.primitives.dsbox.CorexText'},
 'denormalize_step': {'hyperparameters': {},
                      'primitive': 'd3m.primitives.dsbox.Denormalize'},
 'encoder_step': {'hyperparameters': {},
                  'primitive': 'd3m.primitives.dsbox.Encoder'},
 'extract_attribute_step': {'hyperparameters': {'semantic_types': ('https://metadata.datadrivendiscovery.org/types/Attribute',)},
                            'primitive': 'd3m.primitives.data.ExtractColumnsBySemanticTypes'},
 'extract_target_step': {'hyperparameters': {'semantic_types': ('https://metadata.datadrivendiscovery.org/types/Target',
                                                                'https://metadata.datadrivendiscovery.org/types/SuggestedTarget')},
                         'primitive': 'd3m.primitives.data.ExtractColumnsBySemanticTypes'},
 'impute_step': {'hyperparameters': {},
                 'primitive': 'd3m.primitives.sklearn_wrap.SKImputer'},
 'model_step': {'hyperparameters': {},
                'primitive': 'd3m.primitives.sklearn_wrap.SKRandomForestClassifier'},
 'profiler_step': {'hyperparameters': {},
                   'primitive': 'd3m.primitives.dsbox.Profiler'},
 'to_dataframe_step': {'hyperparameters': {},
                       'primitive': 'd3m.primitives.datasets.DatasetToDataFrame'}}
--------------------
[INFO] hit@Candidate: (-3485808742739620201,None)
Traceback (most recent call last):
  File "/nfs1/dsbox-repo/qasemi/dsbox-ta2/python/dsbox/template/search.py", line 372, in setup_initial_candidate
    raise ValueError("Candidate is not compatible with the dataset")
ValueError: Candidate is not compatible with the dataset
[ERROR] Initial Pipeline failed, Trying a random pipeline ...
{'clean_step': {'hyperparameters': {},
                'primitive': 'd3m.primitives.dsbox.CleaningFeaturizer'},
 'corex_step': {'hyperparameters': {},
                'primitive': 'd3m.primitives.dsbox.CorexText'},
 'denormalize_step': {'hyperparameters': {},
                      'primitive': 'd3m.primitives.dsbox.Denormalize'},
 'encoder_step': {'hyperparameters': {},
                  'primitive': 'd3m.primitives.dsbox.Encoder'},
 'extract_attribute_step': {'hyperparameters': {'semantic_types': ('https://metadata.datadrivendiscovery.org/types/Attribute',)},
                            'primitive': 'd3m.primitives.data.ExtractColumnsBySemanticTypes'},
 'extract_target_step': {'hyperparameters': {'semantic_types': ('https://metadata.datadrivendiscovery.org/types/Target',
                                                                'https://metadata.datadrivendiscovery.org/types/SuggestedTarget')},
                         'primitive': 'd3m.primitives.data.ExtractColumnsBySemanticTypes'},
 'impute_step': {'hyperparameters': {},
                 'primitive': 'd3m.primitives.sklearn_wrap.SKImputer'},
 'model_step': {'hyperparameters': {},
                'primitive': 'd3m.primitives.sklearn_wrap.SKRandomForestClassifier'},
 'profiler_step': {'hyperparameters': {},
                   'primitive': 'd3m.primitives.dsbox.Profiler'},
 'to_dataframe_step': {'hyperparameters': {},
                       'primitive': 'd3m.primitives.datasets.DatasetToDataFrame'}}
--------------------
[ERROR] - dsbox.controller.controller - [ERROR] not candidate!

```

**Comment by proska** [comment]
It seems the "utils.remove_columns" method will throw error when all the attributes are categorical and all need to be removed from the input table

**Comment by proska** [comment]
fixed in commit 7cd6460937432efec8a2136adec0391908b9b6c2 on dsbox-cleaning
